Source/player.cpp

```diff
@@ -489,33 +489,35 @@ void NewPlrAnim(int pnum, BYTE *Peq, int numFrames, int Delay, int width, Animat
 	plr[pnum]._pAnimGameTickModifier = 0.0f;
 
 	if (numSkippedFrames != 0 || flags != AnimationFlags::None) {
-		int relevantAnimationLength = numFrames;
+		int relevantAnimationFramesForDistribution = numFrames; // Animation Frames that will be adjusted for the skipped Frames/GameTicks
 		if (distributeFramesBeforeFrame != 0) {
 			// After an attack hits (_pAFNum or _pSFNum) it can be canceled or another attack can be queued and this means the animation is canceled.
 			// In normal attacks frame skipping always happens before the attack actual hit.
 			// This has the advantage that the sword or bow always points to the enemy when the hit happens (_pAFNum or _pSFNum).
 			// Our distribution logic must also regard this behaviour, so we are not allowed to distribute the skipped animations after the actual hit (_pAnimStopDistributingAfterFrame).
-			relevantAnimationLength = distributeFramesBeforeFrame - 1;
+			relevantAnimationFramesForDistribution = distributeFramesBeforeFrame - 1;
 		}
 
+		int gameTicksPerFrame = (Delay + 1);                                                                        // How many GameTicks are needed to advance one Animation Frame
+		int relevantAnimationGameTicksForDistribution = relevantAnimationFramesForDistribution * gameTicksPerFrame; // GameTicks that will be adjusted for the skipped Frames/GameTicks
+
+		int relevantAnimationGameTicksWithSkipping = relevantAnimationGameTicksForDistribution - (numSkippedFrames * gameTicksPerFrame); // How many GameTicks will the Animation be really shown (skipped Frames and GameTicks removed)

```

Comment:
I did this cause then it's more clear that the comment only applies to this specific line.
But I don't mind changing it if this helps.
Question: Only here or should i avoid adding comments at the same line at all occurrences?

Example:
```
// this can happen if we are at the last frame and the next game tick is due (nthread_GetProgressToNextGameTick returns 1.0f)
if (absolutAnimationFrame > relevantAnimationFrames) {
```

	
`	if (absolutAnimationFrame > relevantAnimationFrames) {  // this can happen if we are at the last frame and the next game tick is due (nthread_GetProgressToNextGameTick returns 1.0f)`

Reply:
> Question: Only here or should i avoid adding comments at the same line at all occurrences?

IMHO, always avoid it. Place them above, and leave a blank line above the comment itself to make it clear it is commenting the block below it.

Reply:
Changed it according to your guidance.
And I think in most cases it's more clear.
Thank you 🙂